wg/ws.go

```diff
@@ -136,33 +137,42 @@ func (wswg *WsWgProxy) Port() (int, error) {
 	return udpBindAddr.Port, nil
 }
 
-func (wswg *WsWgProxy) Connect(endpoint string) error {
+func (wswg *WsWgProxy) Connect(ctx context.Context, endpoint string) error {
 	rurl := fmt.Sprintf("wss://%s:443/", endpoint)
 
 	log.Printf("(re-)connecting to %s", rurl)
 
-	conf, _ := websocket.NewConfig(rurl, rurl)
-	conf.TlsConfig = &tls.Config{
-		InsecureSkipVerify: true,
-	}
-
-	// oh well, if it'll end horror
-	ws, err := websocket.DialConfig(conf)
+	ws, _, err := websocket.Dial(ctx, rurl, &websocket.DialOptions{
+		HTTPClient: &http.Client{
+			Transport: &http.Transport{
+				Proxy: http.ProxyFromEnvironment,
+				// It's fine. The traffic inside the tunnel is already encrypted by WG
```

Comment:
Would it have Wireguard over WebSocket over HTTP?

Reply:
> Would it have Wireguard over WebSocket over HTTP?

I think it's fine to have HTTP, but right now backend is listening on HTTPS only and I don't want to touch it at the moment.